### ISSUE TITLE
Update mapping (ES) not working

### ISSUE THREAD

**Issue by bakman2** [zinclabs/zinc#758]
**What were you trying to achieve?**
Update ES mapping - add aggregatable property to a field.

**What action did you take?**
```
PUT /es/homeautomation_test/_mapping 
 
body
{"properties":{"BatteryLevel":{"aggregatable":true,"type":"numeric"}}}
```
**What action/response/output did you expect?**
Expected to add the aggregatable attribute to the field.

**What actually happened?**
Error:
```
index [homeautomation_test] already exists mapping of field [BatteryLevel]
```
**How to reproduce the issue?**
Add a new `"aggregatable": true,` to a field that does not have this property

**What version of ZincSearch are you using?**
0.3.7-9587cdc-nightly

**Any attachments/screenshots?**
<img width="442" alt="image" src="https://user-images.githubusercontent.com/333960/218306191-8e78ed4c-53af-4f0e-8efb-ef8e9b9d43ca.png">

**Which service(s) is this relate to?**
API

### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by bakman2** [comment]
(side)-note: if they are not aggrated, they cannot be used in grafana (somehow).

**Comment by hengfeiyang** [comment]
Because changing the field attribution of `aggregatable` will affect the store behavior, so it doesn't support updates. You need to set the attribute when the field create.

**Comment by bakman2** [comment]
> You need to set the attribute when the field create.

The fields/documents were created with aggregatable set to true, the ES part of the API apperently ignores that ?

```
GET /api/index:

 "shard_num":3,
       "name":"homeautomation_test",
       "storage_type":"disk",
       "settings":{},
       "mappings":{
           "properties":{
               "@timestamp":{
                   "type":"date",
                   "index":true,
                   "store":false,
                   "sortable":true,
                   "aggregatable":true,
                   "highlightable":false
               },
               "BatteryLevel":{
                   "type":"numeric",
                   "index":true,
                   "store":false,
                   "sortable":true,
                   "aggregatable":true,
                   "highlightable":false
```

**Comment by hengfeiyang** [comment]
I'm not got your point. it looks ` "aggregatable":true` already exists for the field `BatteryLevel`. What is the problem?

**Comment by bakman2** [comment]
> I'm not got your point. it looks ` "aggregatable":true` already exists for the field `BatteryLevel`. What is the problem?

Note the difference:

```
/api/index
```
returns `"aggregatable":true,`

vs
```
/es/homeautomation_test/_mapping
```
the ES version does not return `"aggregatable":true,`

**Comment by hengfeiyang** [comment]
@bakman2 Because Elasticsearch Mapping API has no there attribute. the APIs start with `/es/` used to compatible es api.

**Comment by bakman2** [comment]
> @bakman2 Because Elasticsearch Mapping API has no those attributes. the APIs start with `/es/` used to compatible es api.

I am confused. What are you saying exactly?

for use in grafana you need aggregatable metrics, zinc does not expose them on the /es endpoint, why not?

**Comment by hengfeiyang** [comment]
First in the `/es/_target/_mapping` API doesn't response attribute of `aggregatable`, because this API used to compatible es API. in the es API, has no this attribute.

Second, What's you error? What's the error when you use it in grafana.

**Comment by bakman2** [comment]
>Second, What's you error? What's the error when you use it in grafana.

Because it does not expose the aggregatable property, you cannot select a metric and are bound to raw logs only.
In the mean time i uninstalled zinc it doesn't work for me.

**Comment by cnn007** [comment]
> Because changing the field attribution of `aggregatable` will affect the store behavior, so it doesn't support updates. You need to set the attribute when the field create.

There was an error trying to modify mapping title.
![image](https://user-images.githubusercontent.com/13449400/234602263-6e6eff85-2b4f-4a45-ad5f-70fe537389fd.png)

```
"title": {
        "type": "text",
        "index": true,
        "store": false,
        "sortable": false,
        "aggregatable": false,
        "highlightable": false
      }
```
--->
```
{
  "properties": {
    "title": {
      "type": "text",
      "index": true,
      "store": false,
      "sortable": false,
      "aggregatable": false,
      "highlightable": false,
      "analyzer": "gse_search",
      "search_analyzer": "gse_standard"
    }
  }
}
```

**Comment by hengfeiyang** [comment]
Yes, exists field disallow update, like you want to update the analyzer, it need to migrate old data for use new analyzer. or it will cause search error.

So, disallow update exists field.

**Comment by cnn007** [comment]
When will the update operation be used? Can only add new fields.
